File: bible-tools/src/main/java/rawcomposition/bibletools/info/util/glide/PhotoTarget.java
package rawcomposition.bibletools.info.util.glide;

import android.view.View;
import android.widget.ImageView;
import android.widget.ProgressBar;

import com.bumptech.glide.load.resource.drawable.GlideDrawable;
import com.bumptech.glide.request.animation.GlideAnimation;
import com.bumptech.glide.request.target.GlideDrawableImageViewTarget;

import uk.co.senab.photoview.PhotoViewAttacher;

/**
 * Created by tinashe on 2016/11/01.
 */

public class PhotoTarget extends GlideDrawableImageViewTarget {

    private ProgressBar progressBar;

    public PhotoTarget(ImageView view, ProgressBar progressBar) {
        super(view);
        this.progressBar = progressBar;
    }

    @Override
    public void onResourceReady(GlideDrawable resource, GlideAnimation<? super GlideDrawable> animation) {
        super.onResourceReady(resource, animation);
        progressBar.setVisibility(View.GONE);
        PhotoViewAttacher attacher = new PhotoViewAttacher(getView());
        attacher.setScaleType(ImageView.ScaleType.FIT_CENTER);
    }
}

File: bible-tools/src/main/java/rawcomposition/bibletools/info/utils/glide/PhotoTarget.kt
package rawcomposition.bibletools.info.utils.glide

import android.graphics.drawable.Drawable
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.Toast
import com.bumptech.glide.request.target.DrawableImageViewTarget
import com.bumptech.glide.request.transition.Transition
import rawcomposition.bibletools.info.utils.hide

class PhotoTarget constructor(imageView: ImageView,
                              private val progressBar: ProgressBar) : DrawableImageViewTarget(imageView) {

    override fun onResourceReady(resource: Drawable?, transition: Transition<in Drawable>?) {
        super.onResourceReady(resource, transition)

        progressBar.hide()
    }

    override fun onLoadFailed(errorDrawable: Drawable?) {
        super.onLoadFailed(errorDrawable)
        Toast.makeText(getView().context, "load failed!", Toast.LENGTH_SHORT).show()
    }
}